### PR TITLE
+per #18032 Removed final from all persist methods

### PR DESCRIPTION
**Pull request by jgordijn**
This opens PersistentActor for extension.

### CODE REVIEW

**Comment by ktoso** [comment]
As commented in the original thread:

> It would also have to remove `final` from `UntypedPersistentActor` as well as `AbstractPersistentActor`, as well as from `deferAsync` for consistency.
Thanks for picking it up!

**Comment by patriknw** [comment]
Refs #18032

**Comment by patriknw** [comment]
LGTM, please squash to one commit and we will merge it

**Comment by jgordijn** [comment]
@patriknw Done.

**Comment by patriknw** [comment]
OK TO TEST

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 1]
```diff
@@ -350,7 +350,7 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * @param events events to be persisted
    * @param handler handler for each persisted `events`
    */
-  final def persistAllAsync[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit = {
+  def persistAllAsync[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit = {
     events.foreach { event ⇒
       pendingInvocations addLast AsyncHandlerInvocation(event, handler.asInstanceOf[Any ⇒ Unit])
     }
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: This change forgot to include: 
- `persistAsync` on line 361
- `deferAsync` on line 381

**Comment by ktoso** [comment]
LGTM after missing changes added (https://github.com/akka/akka/pull/18148#discussion_r36610913)

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala (hunk 1 above)
Comment (jgordijn) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: good catch. Fixed it.

**Comment by ktoso** [comment]
LGTM, thanks!

**Pull request merged**